apps/server/src/modules/h5p-editor/service/libraryStorage.service.ts

```diff
@@ -31,21 +34,17 @@ export class LibraryStorage implements ILibraryStorage {
 	 * @param filename the requested file
 	 */
 	private checkFilename(filename: string): void {
-		if (path.normalize(filename).startsWith(`..${path.sep}`) || path.isAbsolute(filename)) {
-			throw new Error('Illegal filename');
+		if (/\.\.\//.test(filename)) {
+			throw new H5pError('illegal-filename', { filename }, 400);
+		}
+		if (filename.startsWith('/')) {
+			throw new H5pError('illegal-filename', { filename }, 400);
 		}
 	}
 
-	/**
-	 * Returns the path of the file from the library.
-	 * @param library
-	 * @param filename
-	 * @returns the absolute path to the file
-	 */
-	private getFilePath(library: ILibraryName, filename: string): string {
+	private getS3Key(library: ILibraryName, filename: string) {
 		const uberName = LibraryName.toUberName(library);
-		const filePath = `h5p-libraries/${uberName}/${filename}`;
-		return filePath;
+		return `h5p-libraries/${uberName}/${filename}`;
```

Comment:
see above (return separat variable for better readability)

Reply:
Refactored for better readability